L1Trigger/DTTriggerPhase2/src/MPQualityEnhancerFilterBayes.cc

```diff
@@ -0,0 +1,316 @@
+    return false;
+  if (mp.wi2 != second_mp.wi2 or mp.tdc2 != second_mp.tdc2)
+    return false;
+  if (mp.wi3 != second_mp.wi3 or mp.tdc3 != second_mp.tdc3)
+    return false;
+  if (mp.wi4 != second_mp.wi4 or mp.tdc4 != second_mp.tdc4)
+    return false;
+  if (mp.wi5 != second_mp.wi5 or mp.tdc5 != second_mp.tdc5)
+    return false;
+  if (mp.wi6 != second_mp.wi6 or mp.tdc6 != second_mp.tdc6)
+    return false;
+  if (mp.wi7 != second_mp.wi7 or mp.tdc7 != second_mp.tdc7)
+    return false;
+  if (mp.wi8 != second_mp.wi8 or mp.tdc8 != second_mp.tdc8)
+    return false;
+  return true;
+}
+
+int MPQualityEnhancerFilterBayes::shareSL(metaPrimitive mp, metaPrimitive second_mp) {
+  DTSuperLayerId mpId(mp.rawId);
+  DTSuperLayerId second_mpId(second_mp.rawId);
+
+  int output = 0;
+  if (mpId.wheel() != second_mpId.wheel() || mpId.station() != second_mpId.station() ||
+      mpId.sector() != second_mpId.sector()) {
+    return output;
+  }
+
+  int SL1 = 0;
+  int SL3 = 0;
+
+  int SL1_shared = 0;
+  int SL3_shared = 0;
+
+  if (mp.wi1 != -1 and mp.tdc1 != -1) {
+    ++SL1;
+    if (mp.wi1 == second_mp.wi1 and mp.tdc1 == second_mp.tdc1) {
+      ++SL1_shared;
+    }
+  }
+  if (mp.wi2 != -1 and mp.tdc2 != -1) {
+    ++SL1;
+    if (mp.wi2 == second_mp.wi2 and mp.tdc2 == second_mp.tdc2) {
+      ++SL1_shared;
+    }
+  }
+  if (mp.wi3 != -1 and mp.tdc3 != -1) {
+    ++SL1;
+    if (mp.wi3 == second_mp.wi3 and mp.tdc3 == second_mp.tdc3) {
+      ++SL1_shared;
+    }
+  }
+  if (mp.wi4 != -1 and mp.tdc4 != -1) {
+    ++SL1;
+    if (mp.wi4 == second_mp.wi4 and mp.tdc4 == second_mp.tdc4) {
+      ++SL1_shared;
+    }
+  }
+
+  if (mp.wi5 != -1 and mp.tdc5 != -1) {
+    ++SL3;
+    if (mp.wi5 == second_mp.wi5 and mp.tdc5 == second_mp.tdc5) {
+      ++SL3_shared;
+    }
+  }
+  if (mp.wi6 != -1 and mp.tdc6 != -1) {
+    ++SL3;
+    if (mp.wi6 == second_mp.wi6 and mp.tdc6 == second_mp.tdc6) {
+      ++SL3_shared;
+    }
+  }
+  if (mp.wi7 != -1 and mp.tdc7 != -1) {
+    ++SL3;
+    if (mp.wi7 == second_mp.wi7 and mp.tdc7 == second_mp.tdc7) {
+      ++SL3_shared;
+    }
+  }
+  if (mp.wi8 != -1 and mp.tdc8 != -1) {
+    ++SL3;
+    if (mp.wi8 == second_mp.wi8 and mp.tdc8 == second_mp.tdc8) {
+      ++SL3_shared;
+    }
+  }
+
+  // If the two mp share all hits in a SL, we consider that they share that SL
+  if (SL1_shared == SL1 || SL3_shared == SL3)
+    output = 1;
+
+  return output;
+}
+
+int MPQualityEnhancerFilterBayes::BX(metaPrimitive mp) {
+  int bx;
+  bx = (int)round(mp.t0 / (float)LHC_CLK_FREQ);
+  return bx;
+}
+
+// Is this really needed?
+int MPQualityEnhancerFilterBayes::rango(metaPrimitive mp) {
+  // Correlated
+  if (mp.quality > 5)
+    return 2;
+  // uncorrelated
+  else
+    return 1;
+}
+
+void MPQualityEnhancerFilterBayes::filterCousins(std::vector<metaPrimitive> &inMPaths,
+                                                 std::vector<metaPrimitive> &outMPaths) {
+  // int primo_index = 0;
```

Comment:
Is this commented out code needed? If so, please add another line of comment which explains in which case you plan to uncomment it. Otherwise just remove.